src/frontend/src/flows/recovery/recoveryWizard.ts

```diff
@@ -86,10 +86,20 @@ export const recoveryWizard = async (
 ): Promise<void> => {
   // Here, if the user doesn't have any recovery device, we prompt them to add
   // one.
-  const recoveries = await withLoader(() =>
-    connection.lookupRecovery(userNumber)
+  const [recoveries, lastTimestampShown] = await withLoader(() =>
+    Promise.all([
+      connection.lookupRecovery(userNumber),
+      connection.getLastShownWarningPageTimestamp(),
+    ])
   );
-  if (recoveries.length === 0) {
+  const ONE_WEEK_MILLIS = 7 * 24 * 60 * 60 * 1000;
+  const nowInMillis = Date.now();
+  const oneWeekAgoTimestamp = nowInMillis - ONE_WEEK_MILLIS;
+  const hasNotSeenRecoveryPageLastWeek =
+    (lastTimestampShown ?? 0) < oneWeekAgoTimestamp;
+  if (recoveries.length === 0 && hasNotSeenRecoveryPageLastWeek) {
+    // No need to await for this, we don't need the result.
+    void connection.setShownRecoveryWarningPage();
```

Comment:
Ideally we would still make sure this operation is actually completed, before the II window gets closed...